motoman_driver/src/industrial_robot_client/joint_trajectory_action.cpp

```diff
@@ -754,7 +683,187 @@ bool JointTrajectoryAction::withinGoalConstraints(
   return rtn;
 }
 
+void JointTrajectoryAction::goalCB(JointTractoryActionServer::GoalHandle & gh)
```

Comment:
This method was changed. The previous code was not keeping track of the goals that were received. This was only the case when a full robot trajectory was sent (included all robot groups in one trajectory). This resulted in lost goals, meaning that successful executions were not reported and general misbehavior would occur. The code that was added is just some checking to cancel a previous goal if a new one is requested. In addition, upon completion of a goal, it is set to completed.